torch/jit/_recursive.py

```diff
@@ -329,6 +340,10 @@ def init_fn(script_module):
                 cpp_module._register_parameter(name, orig_value, False)
             elif isinstance(orig_value, torch.jit.Attribute):
                 cpp_module._register_attribute(name, attr_type, orig_value.value)
+            elif isinstance(orig_value, Module):
```

Comment:
This seems wrong. I feel like if this wasn't done before, we are compiling all submodules twice. @suo who understands this logic better.

Reply:
concreteType will record the "moduleInterface" modules as attributes instead of modules, so `concrete_type.get_module_names()` will not get that module and it only get compiled here.